app/assets/javascripts/govuk_publishing_components/analytics-ga4/ga4-core.js

```diff
@@ -256,8 +256,10 @@ window.GOVUK.analyticsGa4 = window.GOVUK.analyticsGa4 || {};
         var totalLinks = 0
         for (var i = 0; i < links.length; i++) {
           var link = links[i]
-          // Only index links that are not search results
-          if (!link.getAttribute('data-ga4-ecommerce-path')) {
+          // Only index links that are not search results or do not have a data-ga4-do-not-index attribute
+          if (link.getAttribute('data-ga4-ecommerce-path') || link.getAttribute('data-ga4-do-not-index') !== null) {
+            continue
+          } else {
```

Comment:
Don't have to make this change if you don't feel it's necessary, but I believe the `else {` is redundant here and you could just have the code standalone, after the if statement.

```JavaScript
if (link.getAttribute('data-ga4-ecommerce-path') || 
  link.getAttribute('data-ga4-do-not-index') !== null) {
  continue
}
  
totalLinks++
link.setAttribute('data-ga4-index', '{"index_link": ' + totalLinks + '}')
  
```

Reply:
Wouldn't having the code outside of the `if` statement make it always run, regardless of the outcome of the `if`? I think you actually want

```Javascript
if (link.getAttribute('data-ga4-ecommerce-path') || link.getAttribute('data-ga4-do-not-index') !== null) { 
   totalLinks++ 
   link.setAttribute('data-ga4-index', '{"index_link": ' + totalLinks + '}')
}
```

Reply:
Ah wait no I've misunderstood what `continue` does, ignore me.

Reply:
We want the link to not have both `data-ga4-ecommerce-path` and `data-ga4-do-not-index` on it in order for an index to be set on the link. 

I might be overthinking it, but if you tried:
```JavaScript
if (!link.getAttribute('data-ga4-ecommerce-path') || !link.getAttribute('data-ga4-do-not-index')) { 
   totalLinks++ 
   link.setAttribute('data-ga4-index', '{"index_link": ' + totalLinks + '}')
}
``` 

the issue is that it would still set the index if one attribute existed. For example, if `data-ga4-do-not-index` existed but `data-ga4-ecommerce-path` didn't, it would still set the index. 

And if you did 
```JavaScript
if (!link.getAttribute('data-ga4-ecommerce-path') && link.getAttribute('data-ga4-do-not-index' === null) {
   totalLinks++ 
   link.setAttribute('data-ga4-index', '{"index_link": ' + totalLinks + '}')
}
```
 then the link would need both attributes to not exist in order for an index to be set on it. This makes sense, but then if the link has `'data-ga4-do-not-index` only, it would still set the index since `data-ga4-ecommerce-path` would also need to be on the link in order to stop the index being set.

This is why I thought the method of using `continue` made sense, since it skips the link if one of the attributes exists, instead of trying to do stuff based on things existing or not existing which was getting complicated!

Please correct me if i'm wrong!

Reply:
Ah, just saw you mentioned you misunderstood - ignore me as well 😃